File: codeforces/1600~1699/1684/d.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0, k = 0;
  cin >> n >> k;

  vector<i64> a(n);
  for (int i = 0; i < n; ++i) cin >> a[i];

  vector<i64> b(n);
  for (int i = 0; i < n; ++i) {
    b[i] = a[i] + i + 1;
  }
  sort(b.rbegin(), b.rend());

  i64 ans = accumulate(a.begin(), a.end(), 0LL);
  for (int i = 0; i < k; ++i) {
    ans -= b[i];
    ans += n;
    ans -= i;
  }
  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1600~1699/1684/d.rs
use std::io;
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let inputs = read_vec::<i32>();
    let n = inputs[0];
    let k = inputs[1];
    let a = read_vec::<i64>();
    let mut b: Vec<i64> = a.iter().enumerate().map(|(i, x)| x + (i as i64) + 1).collect();
    let mut ans: i64 = a.iter().sum();

    b.sort();
    b.reverse();

    for i in 0..(k as i64) {
        ans -= b[i as usize];
        ans += n as i64;
        ans -= i;
    }

    println!("{}", ans);
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
